sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -319,6 +319,11 @@ private static bool TryConvertArgument(IConfiguration configuration, string para
                 return TryConvertFromString(configuration, parameterName, s => new Uri(s), out value);
             }
 
+            if (parameterType == typeof(Guid))
```

Comment:
I guess this is already an existing issue with the Uri parsing above, but should we consider using Guid.TryParse or wrapping the parse in a try/catch so we can continue evaluating other ctors if the configuration does not contain a GUID?